### PR TITLE
[office-js-preview] (Outlook) Add getItemIdAsync

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/OfficeDev/office-js-docs-pr/pull/1051/files
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -10564,6 +10564,56 @@ declare namespace Office {
          * @beta
          */
         getInitializationContextAsync(callback?: (asyncResult: Office.AsyncResult<string>) => void): void;
+        /**
+         * Asynchronously gets the ID of a saved item.
+         *
+         * When invoked, this method returns the item ID via the callback method.
+         * 
+         * **Note**: If your add-in calls `getItemIdAsync` on an item in compose mode in order to get an `itemId` to use with EWS or the REST API,
+         * be aware that when Outlook is in cached mode, it may take some time before the item is actually synced to the server.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: "some time" is too vague. I think user the user would appreciate a rough order of magnitude for this duration (seconds, minutes, hours?)

Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Nit: "actually" is unnecessary. Consider "before the cached item is synced..."

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -10564,6 +10564,56 @@ declare namespace Office {
...
+         * Until the item is synced, using the `itemId` will return an error.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: Consider "Until the item is synced, `itemId` returns an error." It's less wordy.

Also, consider adding the error name here.

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -10564,6 +10564,56 @@ declare namespace Office {
...
+         *
+         * @remarks
+         *
+         * **{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}**: ReadItem
+         * 
+         * **{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}**: Appointment Organizer
+         * 
+         * **Errors**: 
+         * 
+         * - ItemNotSaved: The id can't be retrieved until the item is saved.
```
Comment (AlexJerabek) [inline types/office-js-preview/index.d.ts]: ItemNotSaved should be in code ticks.

**Review by AlexJerabek — COMMENTED** [review]
I added some suggestions to the text. Please review and address them as appropriate.

**Comment by amcasey** [comment]
I'm assuming this work is still in progress.  Please let me know if that's not the case.

File: types/office-js-preview/index.d.ts (hunk 1 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: I'll follow up on this with the devs but if anything this will be updated in a future PR.

**Review by AlexJerabek — APPROVED** [review]

**Pull request merged**